drivers/mge-utalk.c

```diff
@@ -896,19 +896,17 @@ static int mge_command(char *reply, int replylen, const char *fmt, ...)
 	}
 
 	/* send terminating string */
-	if (MGE_COMMAND_ENDCHAR) {
-		for (p = MGE_COMMAND_ENDCHAR; *p; p++) {
-			if ( isprint(*p & 0xFF) )
-				upsdebugx(4, "mge_command: sending [%c]", *p);
-			else
-				upsdebugx(4, "mge_command: sending [%02X]", *p);
+	for (p = MGE_COMMAND_ENDCHAR; *p; p++) {
```

Comment:
@aquette @clepple @zykh : experts wanted :) In current codebase, `MGE_COMMAND_ENDCHAR` is `#define`d above as `"\r\n"`. Are there variations of the codebase with a `NULL` instead? Is there some experimental, device-dependent, etc. need for such cases? The original (removed by PR) check seems to relate to that - "if we have a string, send it" - and with the `if` removed we'd get into NULL-pointer dereferencing by `for ... *p`.

Probably both goals could be achieved by `for (p = MGE_COMMAND_ENDCHAR; p && *p; p++) {...}` if there is a chance for builds with a `MGE_COMMAND_ENDCHAR==NULL` ?

- [ ] Revise dropping of NULLness check

Reply:
good to me.
Not sure why we put that?! part of the very early work, back around 2001